web-marketplace/src/clients/Clients.SharedProviders.tsx

```diff
@@ -0,0 +1,117 @@
+import { IntercomProvider } from 'react-use-intercom';
+import amplitudePlugin from '@analytics/amplitude';
+import googleAnalytics from '@analytics/google-analytics';
+import { Box } from '@mui/material';
+import { Theme } from '@mui/material/styles';
+import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFns';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import * as Sentry from '@sentry/react';
+import { BrowserTracing } from '@sentry/tracing';
+import { QueryClientProvider } from '@tanstack/react-query';
+import { ReactQueryDevtools } from '@tanstack/react-query-devtools';
+import Analytics from 'analytics';
+import doNotTrack from 'analytics-plugin-do-not-track';
+import { AnalyticsProvider } from 'use-analytics';
+
+import ThemeProvider from 'web-components/src/theme/RegenThemeProvider';
+
+import { PropsWithChildren } from 'types/react/children';
+import { apolloClientFactory } from 'lib/clients/apolloClientFactory';
+import { reactQueryClient } from 'lib/clients/reactQueryClient';
+
+import PageLoader from 'components/atoms/PageLoader';
+
+import { AuthApolloProvider } from '../apollo';
+
+const intercomId = import.meta.env.VITE_INTERCOM_APP_ID || '';
+
+// by default do not enable sentry unless the flag is set.
+// this reduces our monthly usage which can quickly run out if this is not set.
+// you can set this flag in local environments if testing changes to sentry.
+// currently we only want this flag set for the production environment.
+if (import.meta.env.VITE_SENTRY_ENABLED) {
+  const defaultSampleRate = '0.2';
+  const tracesSampleRate = parseFloat(
+    import.meta.env.VITE_SENTRY_TRACES_SAMPLE_RATE || defaultSampleRate,
+  );
+  Sentry.init({
+    dsn: 'https://f5279ac3b8724af88ffb4cdfad92a2d4@o1377530.ingest.sentry.io/6688446',
+    integrations: [
+      new BrowserTracing({
+        routingInstrumentation: Sentry.reactRouterV6Instrumentation(
+          useEffect,
+          useLocation,
+          useNavigationType,
+          createRoutesFromChildren,
+          matchRoutes,
+        ),
+      }),
+    ],
+    tracesSampleRate,
+    environment: import.meta.env.VITE_SENTRY_ENVIRONMENT || 'development',
+  });
+}
+
+// our current analytics setup uses both amplitude and google analytics.
+// our amplitude and GA have been set up with a development and production environment.
+// for amplitude this means that we have a development and production API key.
+// for GA this means that we have a development and production measurement id.
+// be careful not to use the production API key or measurement id in a non-prod environment.
+// but you can safely use the development API key and measurement id in non-prod environments.
+// these values are safe to hardcode because they are public.
+// also see the VITE_ANALYTICS_ENABLED environment variable.
+const DEVELOPMENT_AMPLITUDE_API_KEY = 'ef9a9d58cf90476430f62a634d72cd5c';
+const DEVELOPMENT_GA_MEASUREMENT_ID = 'G-9ENS4JTCWY';
```

Comment:
it would make sense to get rid of those default values now, even in dev, tracking for different client apps should be different